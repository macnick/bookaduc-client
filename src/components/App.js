import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';

import Navbar from './layout/Navbar';
import Home from './layout/Home';
import BikeList from '../containers/BikeList';
import BikeDetails from './layout/BikeDetails';

import Login from './auth/Login';
import SignUp from './auth/SignUp';

const App = () => (
  <BrowserRouter>
    <div className="app">
      <Navbar />
      <Switch>
        <Route path="/bikes/:id" component={BikeDetails} />
        <Route path="/bikes/" component={BikeList} />
        <Route path="/login" component={Login} />
        <Route path="/signup" component={SignUp} />

        <Route path="/" component={Home} />
      </Switch>
    </div>
  </BrowserRouter>
);

export default App;
