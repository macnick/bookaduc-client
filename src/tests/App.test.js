import React from 'react';
import { Provider } from 'react-redux';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import App from '../components/App';
import '@testing-library/jest-dom';

const mockStore = configureStore([]);

describe('Renders my Redux connected component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      auth: {
        token:
          'eyJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjo0OCwiZXhwIjoxNjE2MzI0Njk3fQ.b5kw-uh4TqYdDizAHq3z_KMvZj3i-mjznMpmxN1fb1M',
        authStatus: true,
        userId: '48',
        loading: false,
        error: null,
      },
      book: {
        user_id: 48,
        bookings: [
          {
            id: 140,
            city: 'Athens',
            date: '17/03/2021',
            bike: {
              id: 1,
              name: 'Monster 1200S',
            },
          },
        ],
        loading: false,
        name: 'Batman',
      },
      bikes: {
        bikes: [
          {
            id: 1,
            name: 'Monster 1200S',
            displacement: '1198 cc',
            power: '145 hp',
            torque: '124 Nm',
            weight: '185 Kg',
            image:
              'https://media.ducati.com/images/previews/mon/m1200s/m1200s-gbg-light.png?resize=1024:*',
            bookings: [
              {
                id: 103,
                date: '2020-12-26',
                city: 'Athens',
                user: {
                  id: 47,
                  name: 'Test',
                },
              },
            ],
          },
          {
            id: 2,
            name: 'SuperSport',
            displacement: '937 cc',
            power: '110 hp',
            torque: '93 Nm',
            weight: '184 Kg',
            image:
              'https://media.ducati.com/images/previews/ss1/ss939s/ss939s-r-light.png?resize=1024:*',
            bookings: [
              {
                id: 94,
                date: '2020-10-31',
                city: 'Rome',
                user: {
                  id: 26,
                  name: 'Superman',
                },
              },
            ],
          },
        ],
        loading: true,
      },
    });

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });

  afterEach(cleanup);

  it('should render the App', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  // it('should render the Header', () => {
  //   const { getByText } = render(<Header />);
  //   expect(getByText(/Book A Ducati/)).toBeInTheDocument();
  // });

  // it('should render the selection menu', () => {
  //   const { getByText } = render(
  //     <Provider store={store}>
  //       <App />
  //     </Provider>
  //   );
  //   expect(getByText(/Select Country/)).toBeInTheDocument();
  // });
});
