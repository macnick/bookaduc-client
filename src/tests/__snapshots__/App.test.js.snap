// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders my Redux connected component should render the App 1`] = `
<div
  className="app"
>
  <div
    className="container section"
  >
    <ul
      className="sidenav right"
      id="menu-side"
    >
      <li>
        <a
          aria-current="page"
          className="sidenav-close active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          <i
            className="material-icons"
          >
            person
          </i>
          Batman
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          className="sidenav-close"
          href="/user"
          onClick={[Function]}
        >
          <i
            className="material-icons"
          >
            date_range
          </i>
          Appointments
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          className="sidenav-close"
          href="/bikes"
          onClick={[Function]}
        >
          <i
            className="material-icons"
          >
            motorcycle
          </i>
          Bikes
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          className="sidenav-close"
          href="/logout"
          onClick={[Function]}
        >
          <i
            className="material-icons"
          >
            logout
          </i>
          Log Out
        </a>
      </li>
    </ul>
    <button
      className="sidenav-trigger"
      data-target="menu-side"
      href="#"
      type="button"
    >
      <i
        className="material-icons small"
      >
        menu
      </i>
    </button>
  </div>
  <div
    className="dashboard "
    id="bikelist"
  >
    <div
      className="react-multi-carousel-list  "
    >
      <ul
        className="react-multi-carousel-track "
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "overflow": "unset",
            "transform": "translate3d(0px,0,0)",
            "transition": "none",
          }
        }
      />
    </div>
  </div>
</div>
`;
